### PR TITLE
Fix: Cached AccessToken missing realm property in OIDC scenario

### PR DESCRIPTION
**Pull request by bbush915**
In the case an application is using a separate OIDC-compliant authority and the tid claim is empty, the cached access token does not contain a realm property. One side-effect is that the cached access token will fail the isAccessTokenEntity check, preventing it from being re-used.

### CODE REVIEW

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 1]
```diff
@@ -468,7 +468,9 @@ export class ResponseHandler {
                 env,
                 serverTokenResponse.access_token,
                 this.clientId,
-                idTokenClaims?.tid || authority.tenant,
+                idTokenClaims?.tid ||
+                    authority.tenant ||
+                    Constants.EMPTY_STRING,
```
Comment (tnorling) [inline lib/msal-common/src/response/ResponseHandler.ts]:
```suggestion
                idTokenClaims?.tid ||
                    authority.tenant ||
                    "",
```

**Review by tnorling — CHANGES_REQUESTED** [review]
Thanks for the PR! Can you please add a test & generate changefiles by running `npm run beachball:change` from the root of the repo.

**Comment by bbush915** [comment]
@microsoft-github-policy-service agree [company="Dialexa, an IBM Company"]

**Comment by bbush915** [comment]
@microsoft-github-policy-service agree company="Dialexa, an IBM Company"

File: lib/msal-common/src/response/ResponseHandler.ts (hunk 1 above)
Comment (bbush915) [reply lib/msal-common/src/response/ResponseHandler.ts]: Updated!

**Comment by bbush915** [comment]
@tnorling addressed comment and added a test. The beachball command reported no change files are needed.

**Comment by tnorling** [comment]
> @tnorling addressed comment and added a test. The beachball command reported no change files are needed.

Ah I suspect this is because your feature branch is also called dev. Can you try another branch?

**Comment by bbush915** [comment]
> > @tnorling addressed comment and added a test. The beachball command reported no change files are needed.
> 
> Ah I suspect this is because your feature branch is also called dev. Can you try another branch?

Good call! Sorry for the messy commits, but I got the change files in place!

**Review by tnorling — APPROVED** [review]

**Comment by bbush915** [comment]
@tnorling Hello! Is there anything I can do to help this land?

**Comment by sameerag** [comment]
@bbush915 I can help land this. Looks like some tests are failing.

**Comment by bbush915** [comment]
@sameerag I validated that tests are all passing locally. Is there anything I can do to troubleshoot the failing E2E tests? It looks like they are all reporting: `Git fetch failed with exit code: 128`

**Comment by bbush915** [comment]
@tnorling @sameerag following up on the above, I'm not sure why the E2E tests are failing, but can do what is needed to assist. Hoping to get this to land soon, since it is blocking some upgrades.

**Comment by BboyAkers** [comment]
Hey cool and awesome former colleagues 🙂! Can this get approved and merged? It's blocking a couple companies and their clients including one I'm helping with 😅🙏🏾. So sorry, for the push and hope I don't come off as rude!

**Review by sameerag — APPROVED** [review]

**Comment by sameerag** [comment]
> Hey cool and awesome former colleagues 🙂! Can this get approved and merged? It's blocking a couple companies and their clients including one I'm helping with 😅🙏🏾. So sorry, for the push and hope I don't come off as rude!

Approved and updated the base branch. Hopefully tests pass now.

**Comment by BboyAkers** [comment]
> > Hey cool and awesome former colleagues 🙂! Can this get approved and merged? It's blocking a couple companies and their clients including one I'm helping with 😅🙏🏾. So sorry, for the push and hope I don't come off as rude!
> 
> Approved and updated the base branch. Hopefully tests pass now.

 😭 🙏🏾 ❤️
[image](https://media.tenor.co/images/3ed84016385a06560957920423fea2f8/raw)

**Pull request merged**